packages/dev/core/src/Misc/textureTools.ts

```diff
@@ -230,6 +231,179 @@ export function FromHalfFloat(value: number): number {
     return (s ? -1 : 1) * Math.pow(2, e - 15) * (1 + f / Math.pow(2, 10));
 }
 
+const ProcessAsync = async (
+    texture: BaseTexture,
+    width: number,
+    height: number,
+    face: number,
+    channels: { R: boolean; G: boolean; B: boolean; A: boolean },
+    lod: number,
+    resolve: (result: Uint8Array) => void,
+    reject: () => void
+): Promise<void> => {
+    const scene = texture.getScene()!;
+    const engine = scene.getEngine();
+
+    let lodPostProcess: PostProcess;
+
+    if (!texture.isCube) {
+        lodPostProcess = new PostProcess("lod", "lod", ["lod", "gamma"], null, 1.0, null, Texture.NEAREST_NEAREST_MIPNEAREST, engine);
+    } else {
+        const faceDefines = ["#define POSITIVEX", "#define NEGATIVEX", "#define POSITIVEY", "#define NEGATIVEY", "#define POSITIVEZ", "#define NEGATIVEZ"];
+        lodPostProcess = new PostProcess("lodCube", "lodCube", ["lod", "gamma"], null, 1.0, null, Texture.NEAREST_NEAREST_MIPNEAREST, engine, false, faceDefines[face]);
+    }
+
+    if (!lodPostProcess.getEffect().isReady()) {
+        // Try again later
+        lodPostProcess.dispose();
+
+        setTimeout(() => {
+            ProcessAsync(texture, width, height, face, channels, lod, resolve, reject);
+        }, 250);
+
+        return;
+    }
+
+    const rtt = new RenderTargetTexture("temp", { width: width, height: height }, scene, false);
+
+    lodPostProcess.onApply = function (effect) {
+        effect.setTexture("textureSampler", texture);
+        effect.setFloat("lod", lod);
+        effect.setBool("gamma", texture.gammaSpace);
+    };
+
+    const internalTexture = texture.getInternalTexture();
+
+    if (rtt.renderTarget && internalTexture) {
+        const samplingMode = internalTexture.samplingMode;
+        if (lod !== 0) {
+            texture.updateSamplingMode(Texture.NEAREST_NEAREST_MIPNEAREST);
+        } else {
+            texture.updateSamplingMode(Texture.NEAREST_NEAREST);
+        }
+
+        scene.postProcessManager.directRender([lodPostProcess], rtt.renderTarget, true);
+        texture.updateSamplingMode(samplingMode);
+
+        // Read the contents of the framebuffer
+        const numberOfChannelsByLine = width * 4;
+        const halfHeight = height / 2;
+
+        //Reading datas from WebGL
+        const bufferView = await engine.readPixels(0, 0, width, height);
+        const data = new Uint8Array(bufferView.buffer, 0, bufferView.byteLength);
+
+        if (!channels.R || !channels.G || !channels.B || !channels.A) {

```

Comment:
This block of code seems quite specific to the inspector...

Maybe we could make `channels` an optional parameter and skip this block if not provided?

Reply:
this function is the private implementation but the in the public function below, the channels default to all true so this code block would generally be skipped.